src/lib/grpc-devtools.ts

```
@@ -0,0 +1,87 @@
+// Taken from this comment on github:timostamm/protobuf-ts:
+// https://github.com/timostamm/protobuf-ts/issues/584#issuecomment-1735664168
+
+import { RpcError, type RpcInterceptor } from "@protobuf-ts/runtime-rpc";
+
+const type = "__GRPCWEB_DEVTOOLS__";
+
+export const grpcWebDevToolsInterceptor: RpcInterceptor = {
+    interceptUnary(next, method, input, options) {
+        const res = next(method, input, options);
+        // @ts-expect-error Variable injected by grpc-web-devtools in the browser
```

I know that this is the exactly solution posted in the comment you mentioned, but this doesn't always mean that it is the best.

To simply ignore the error is a solution but there is another way.

```
declare global {
    interface Window {
        __GRPCWEB_DEVTOOLS__?: boolean;
    }
}
```

By declaring this we can extend the global Window object without causing conflicts in the global namespace.

This snippet can either go in this file or into a more appropriate space like `src/types/global.d.ts` or any other file where we already have types (but I don't think we have a file already).

---

I think putting it at the top of the `grpc-devtools.ts` file is where it belongs. I've added it and removed the ts-expect-error comments.

---

Perfect. I also think this can stay in this file.